File: PHP/program/Clothing.php
<!-- Kelas Clothing -->
<?php

require('Product.php'); //parent

class Clothing extends Product{
    private $size = '';
    private $material = '';
    private $gender = '';

    // Constructor
    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender) {
        parent::__construct($idProduct, $name, $brand, $price);
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    // Setter untuk size
    public function setSize($size) {
        $this->size = $size;
    }

    // Getter untuk size
    public function getSize() {
        return $this->size;
    }

    // Setter untuk material
    public function setMaterial($material) {
        $this->material = $material;
    }

    // Getter untuk material
    public function getMaterial() {
        return $this->material;
    }

    // Setter untuk gender
    public function setGender($gender) {
        $this->gender = $gender;
    }

    // Getter untuk gender
    public function getGender() {
        return $this->gender;
    }
}
?>

File: CPP/program/Clothing.cpp
//import library
#include <iostream>
#include <string>
#include "Product.cpp"

//using standar namespace
using namespace std;

class Clothing : public Product{ //class Clothing

    //Private attributes
    private:
        string size; 
        string material;        
        string gender;
        
    //public methods
    public:
        Clothing(){
            this->size = "";
            this->material = "";
            this->gender = "";
        }

        Clothing(string size, string material, string gender){
            this->size = size;
            this->material = material;
            this->gender = gender;
        }

        string get_size() { return size; }
        string get_material() { return material; }
        string get_gender() { return gender; }

        void set_size(string size) {
        this->size = size; 
        }
        void set_material(string material) {
        this->material = material; 
        }
        void set_gender(string gender) {
        this->gender = gender; 
        }

        //destractor
        ~Clothing(){
            
        }
};